design/operator-architecture.asciidoc

```diff
@@ -0,0 +1,154 @@
+# Keycloak Operator High Level Architecture
+
+* **Status**: Draft #1
+* **JIRA**: https://issues.redhat.com/browse/KEYCLOAK-10319[KEYCLOAK-10319]
+* **Source of the images**: https://docs.google.com/presentation/d/13N5ClXXXcxKjXgor72rUdtMBN07_GneUkhTTwBsy_tE/edit?usp=sharing[Google Slides]
+
+This document shows a big picture how all the components and layers of Keycloak Operator work together. The blueprint presented here shall be used a guideline for designing features as well as integrating community Pull Requests.
+
+## Manifesto
+
+We envision Keycloak Operator as:
+
+* an opinionated way of installing and maintaining Keycloak installation on both Kubernetes and OpenShift.
+* a way to deliver a high level usability rather than low-level toggles
+* an example of a recommended way of installing, configuring, managing custom extension and themes.
+* the thinnest possible layer, with the least possible logic, on top of Keycloak Container and Keycloak Server.
+
+## Use cases
+
+Our vision for Keycloak Operator is driven by use cases. The use cases mention the following personas:
+
+* Cluster Admin - a Kubernetes/OpenShift administrator
+* Service Infrastructure Admin - an administrator who provisions Keycloak Operator as well as Keycloak Cluster
+* Realm Admin - a Keycloak administrator, who can provision realms (both using Custom Resources and Admin UI)
+* Secured Application - either configured by an engineering team (using self registration feature) or by Keycloak Admin (or Keycloak Realm Admin).
```

Comment:
I'm not fully convinced about these personas. Looking at the doc I sent you there are:

- Cluster Admin
- Service Infrastructure Admin
- Service Admin
- Service Consumer
- Service End User

I think those are better personas then those you've created. From my perspective the description in context of the Keycloak Operator would be something like:

- Cluster Admin - Provisions cluster resources. Basically, installs the Keycloak Operator.
- Service Infrastructure Admin - Provisions the managed services. Basically, creates the Keycloak CR.
- Service Admin - Manages the Realm. Basically, create the KeycloakRealm CR, but can also manage it through the Keycloak admin console
- Service Consumer - Application developers. Basically, creates the KeycloakClient CR, but can also create clients through other means (admin console, client registration cli, etc.)
- Service End User - Application users. Not sure there are any use-cases for these in this context, but worth mentioning anyways